### PR TITLE
[FEATURE] Add Dependency Ramp Manager to support System Dependency Ra…

### PR DESCRIPTION
**Pull request by kxu1026**
…mpping Feature

### CODE REVIEW

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java [hunk 1]
```diff
@@ -0,0 +1,756 @@
...
+      for (ExecutableRamp executableRamp : executableRampMap.getActivatedAll()) {
+        try {
+          String rampId = executableRamp.getId();
+
+          // get Base Props
+          Props baseProps = new Props();
+          baseProps.putAll(executableRampDependencyMap.getDefaultValues(executableRampItemsMap.getDependencies(rampId)));
+
+          ExecutableRampStatus status = executableRampExceptionalFlowItemsMap.check(rampId, flowName);
+          switch (status) {
```
Comment (jamiesjc) [inline azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java]: Can the switch clause here be simplified?

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java [hunk 2]
```diff
@@ -0,0 +1,756 @@
...
+  /**
+   * Call to set Executable Ramp Metadata into ExecutableFlow
+   */
+  synchronized public void configure(ExecutableFlow executableFlow, File flowDirectory) {
+
+    if (isRampFeatureEnabled) {
+
+      // To be safe, check if there is any jar files in ./excluded folder
+      // and move them back to the place in original location of the package
+      moveFiles(
```
Comment (jamiesjc) [inline azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java]: Can you explain why the jar files need to be moved?

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java [hunk 3]
```diff
@@ -0,0 +1,756 @@
...
+
+    public void start() {
+      this.scheduler.scheduleAtFixedRate(() -> pollExecution(), 0L, this.statusPollingIntervalMinutes,
+          TimeUnit.MINUTES);
+    }
+
+    private void pollExecution() {
+      if (this.pollingCriteria.shouldPoll()) {
+        if (this.pollingCriteria.satisfiesUnsavedDataAvailableCriteria()) {
+          FlowRampManager.this.saveSettings();
```
Comment (jamiesjc) [inline azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java]: Why does `FlowRampManager` also need to call `pollExecution` periodically?

**Comment by juhoautio** [comment]
Is there any high level description of what this new "rampping" feature does anywhere? Not to mention actual documentation.. I couldn't find anything when searching "ramp" in https://azkaban.readthedocs.io/en/latest/.

For now, could someone briefly explain what the "ramp" does?

**Pull request merged**